src/app/donation-start/donation-start-form/donation-tipping-slider/donation-tipping-slider.component.ts

```diff
@@ -0,0 +1,214 @@
+  // the calculation results from input changes
+   else if (!this.disableDefaults){
+      if (!this.donationAmount) {
+        this.derivedPercentage = 0;
+      }
+      if (this.donationAmount >= 1000) {
+        this.derivedPercentage = 7.5;
+      } else if (this.donationAmount >= 300) {
+        this.derivedPercentage = 10;
+      } else if (this.donationAmount >= 100) {
+        this.derivedPercentage = 12.5;
+      } else {
+        this.derivedPercentage = 15;
+      }
+    }
+  }
+
+  format(currencyCode: 'GBP' | 'USD', amount: number) {
+    return Intl.NumberFormat('en-GB', {
+      style: 'currency',
+      currency: currencyCode,
+    }).format(amount);
+  }
+
+  move = (e: MouseEvent | TouchEvent) => {
+    if (this.isMoving) {
+      this.disableDefaults = true;
+      this.max = this.bar.nativeElement.offsetWidth - this.handle.nativeElement.offsetWidth;
+
+      if (window.TouchEvent && e instanceof TouchEvent) {
+        this.pageX = e.touches[0]?.pageX;
+      } else {
+        // we Know e is a MouseEvent because all platforms that supports TouchEvent would also have
+        // a truthy window.TouchEvent - see https://stackoverflow.com/a/32882849/2803757
+        this.pageX = (e as MouseEvent).pageX;
+      }
+
+      if (this.pageX !== undefined) {
+        if(this.derivedPercentage) {
+          this.calcAndSetTipAmount();
+          this.updateHandlePosition(e);
+          this.adjustDonationPercentageAndValue();
+          this.onHandleMoved(this.derivedPercentage, this.tipAmount);
+        }
+      }
+    }
+
+  }
+
+  calcAndSetTipAmount() {
+    if (this.donationAmount < 55) {
+      this.tipAmount = this.donationAmount * (this.derivedPercentage / 100);
+    } else {
+      this.tipAmount = Math.round(this.donationAmount * (this.derivedPercentage / 100));
+    }
+  }
+
+  updateHandlePosition(e: MouseEvent | TouchEvent | undefined) {
+    if(!e) { // the handle position is driven by the donation input change
+      this.calcAndSetPercentage();
+      this.pageX = 122;
+      this.mousePos = this.pageX - this.bar.nativeElement.offsetLeft - this.handle.nativeElement.offsetWidth / 2;
+      this.position = this.max * this.derivedPercentage / this.percentageEnd;
+      this.handle.nativeElement.style.marginLeft = this.position + 'px';
+    } else { // the handle position is driven by the mouse click on slider
+      this.mousePos = this.pageX - this.bar.nativeElement.offsetLeft - this.handle.nativeElement.offsetWidth / 2;
+      this.position = this.mousePos > this.max ? this.max : this.mousePos < 0 ? 0 : this.mousePos;
+      this.calcAndSetPercentage();
+    }
+
+    this.handle.nativeElement.style.marginLeft = this.position + 'px';
+  }
+
+  adjustDonationPercentageAndValue() {
+    this.percentageWrap.nativeElement.innerHTML = this.derivedPercentage.toString();
+    this.currencyFormatted = this.format(this.donationCurrency, this.tipAmount);
+    this.donationWrap.nativeElement.innerHTML = this.currencyFormatted;
```

Comment:
can we set `.text` instead of `.innerHtml`, since neither the percentage or the currency are HTML snippets?

And I guess we don't need to update the currencyFormatted do we?